### PR TITLE
[FSSDK-9778] return last experiment when duplicate key in config

### PR DESCRIPTION
**Pull request by raju-opti**
## Summary
When duplicate experiment keys are found in a datafile, SDK returns a correct experimentMap in OptimizelyConfig:
1. the experimentMap will contain the experiment later in the datafile experiments list.
2. add a warning log about "Duplicate experiment keys found in datafile: {key-name}"

## Test plan
- add a test for config with duplicate experiment keys

## Issues
- FSSDK-9778

### CODE REVIEW

File: lib/core/optimizely_config/index.ts [hunk 1]
```diff
@@ -347,39 +351,52 @@ export class OptimizelyConfig {
    * @param       {ProjectConfig}                           configObj
    * @param       {FeatureVariablesMap}                     featureIdVariableMap
    * @param       {{[id: string]: FeatureVariable}}         variableIdMap
-   * @returns     {[id: string]: OptimizelyExperiment}      Experiments mapped by id
+   * @returns     { experimentsMapById: { [id: string]: OptimizelyExperiment }, experimentsMapByKey: OptimizelyExperimentsMap }      Experiments mapped by id and key
    */
-  static getExperimentsMapById(
+  static getExperimentsMap(
```
[the remark this answers is not in the document]
Comment (raju-opti) [reply lib/core/optimizely_config/index.ts:356]: I thing there shouldn't be any concern
